src/rail/estimation/algos/lephare.py

```diff
@@ -1,22 +1,155 @@
+        # Get location to store filter files
+        filter_output = os.path.join(
+            os.environ["LEPHAREWORK"], "filt", self.lephare_config["FILTER_FILE"].value
+        )
+        # Write filter files
+        lp.write_output_filter(
+            filter_output + ".dat", filter_output + ".doc", filterLib
+        )
+
+    def _create_sed_library(self):
+        """Make the SED binary library files in lephare format.
+
+        We separately create the star, quasar and galaxy libraries.
+        """
+        sedlib = lp.Sedtolib(config_keymap=self.lephare_config)
+        sedlib.run(typ="STAR", star_sed="$LEPHAREDIR/examples/STAR_MOD_ALL.list")
+        sedlib.run(typ="QSO", qso_sed="$LEPHAREDIR/sed/QSO/SALVATO09/AGN_MOD.list")
+        sedlib.run(typ="GAL", gal_sed="$LEPHAREDIR/examples/COSMOS_MOD.list")
+
+    def _create_mag_library(self):
+        """Make the magnitudes library file in lephare format.
+
+        We separately create the star, quasar and galaxy libraries.
+
+        TODO: replace hardcoded config options with class config options.
+        """
+        maglib = lp.MagGal(config_keymap=self.lephare_config)
+        maglib.run(typ="STAR", lib_ascii="YES")
+        maglib.run(
+            typ="QSO",
+            lib_ascii="YES",
+            mod_extinc="0,1000",
+            eb_v="0.,0.1,0.2,0.3",
+            extinc_law="SB_calzetti.dat",
+        )
+        maglib.run(
+            typ="GAL",
+            lib_ascii="YES",
+            mod_extinc="18,26,26,33,26,33,26,33",
+            extinc_law=(
+                "SMC_prevot.dat,SB_calzetti.dat,"
+                + "SB_calzetti_bump1.dat,SB_calzetti_bump2.dat"
+            ),
+            em_lines="EMP_UV",
+            em_dispersion="0.5,0.75,1.,1.5,2.",
+        )
+
+    def run(self):
+        """Run rail_lephare inform stage.
+
+        This is the basic informer which takes the config and templates and
+        makes the inputs required for the run.
+
+        In addition to the three lephare stages making the filter, sed, and
+        magnitude libraries we also do some tasks required by all rail inform
+        stages.
+        """
+        # Set training data required for all informers?
+        if self.config.hdf5_groupname:
+            training_data = self.get_data("input")[self.config.hdf5_groupname]
+        else:  # pragma: no cover
+            training_data = self.get_data("input")
+
+        # Get number of sources
+        ngal = len(training_data[self.config.ref_band])
+
+        # The three main lephare specific inform tasks
+        self._create_filter_library()
+        self._create_sed_library()
+        self._create_mag_library()
+
+        # Spectroscopic redshifts
+        self.szs = training_data[self.config.redshift_col]
+
+        # Give principle inform config 'model' to instance.
+        self.model = dict(config_file=self.config_file)
+        self.add_data("model", self.model)
```

Comment:
Is this what BPZ is doing? I'm not criticizing it, just curious. Ceci likes there to be some output file from each stage that it seems to use as an indicator that the stage has completed, and that it's ok to move to the next stage, so perhaps this fulfills that need.